lib/ui/painting.dart

```diff
@@ -39,6 +39,13 @@ bool _offsetIsValid(Offset offset) {
   return true;
 }
 
+bool _matrix4IsValid(Float64List matrix4) {
+  assert(matrix4 != null, 'Matrix4 argument was null.');
+  if (matrix4.length != 16)
+    throw new ArgumentError('"matrix4" must have 16 entries.');
```

Comment:
Since you only call this from asserts, maybe just make this an assert (like the line above) rather than throwing an explicit exception